index.js

```diff
@@ -516,13 +516,23 @@ Line2D.prototype.update = function (options) {
 					pos[ptr++] = y
 				}
 
-				let triangles = triangulate(pos, state.hole || [])
-
-				for (let i = 0, l = triangles.length; i < l; i++) {
-					if (ids[triangles[i]] != null) triangles[i] = ids[triangles[i]]
+				if(!(state.count-1 in ids))
+					ids[state.count] = state.count-1   // make sure there is at least polygon
+					
+				let splits = Object.keys(ids).map(e => e|0).sort(function(a, b){return a - b})
+				
+				let split_triangles = []
+				let base = 0
+				
+				for (let i = 0; i < splits.length; i++)
+				{
+					let triangles = triangulate(pos.slice(base*2, splits[i]*2), state.hole || [])
```

Comment:
I am not sure I follow the logic here, but placing triangulate in a loop looks suspicious.
Could you please elaborate a bit what algorithmic changes are done here?

Reply:
The issue is when plotly send scatter segment to regl (which is seperated by null in plotly input), plotly close each segment and drop the nulls. When regl receive the segments, it treat it as a single big plygon which end up drawing some extra triangles.

I submitted a PR (https://github.com/plotly/plotly.js/pull/5355) to plotly to keep those nulls. However that it self is not enough since regl will replace the null with last valid point and still send a single polygon for triangulation. I don't see earcut.js supporting multiple polygon in a single input yet. So the proposed solution here is to call triangulation multiple time, each time only handle one segment (polygon).